symfony/framework-bundle/4.2/public/index.php

```diff
@@ -7,28 +7,26 @@
 
 require __DIR__.'/../vendor/autoload.php';
 
-// The check is to ensure we don't use .env if APP_ENV is defined
-if (!isset($_SERVER['APP_ENV'])) {
-    if (!class_exists(Dotenv::class)) {
-        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
-    }
-    (new Dotenv())->load(__DIR__.'/../.env');
+if (class_exists(Dotenv::class)) {
+    (new Dotenv())->loadForEnv($env ?? 'dev', __DIR__.'/../.env');
+} elseif (!isset($_SERVER['APP_ENV']) && !isset($_ENV['APP_ENV'])) {
+    throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
 }
 
-$env = $_SERVER['APP_ENV'] ?? 'dev';
-$debug = (bool) ($_SERVER['APP_DEBUG'] ?? ('prod' !== $env));
+$env = $_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? 'dev';
+$debug = (bool) ($_SERVER['APP_DEBUG'] ?? $_ENV['APP_DEBUG'] ?? ('prod' !== $env));
```

Comment:
Could we centralize some of this repeated logic between `index.php` and `console? Specifically, the loading of the DotEnv stuff and this logic. It *must* be consistent between console/index.php & testing. So I think it's important.

Reply:
done, I just moved `bootstrap.php` in `src/` and used it in `index.php` and `console`.